README.md

```diff
@@ -18,211 +21,185 @@
+Add `PostgresNIO` as dependency to your `Package.swift`:
 
 ```swift
-import PostgresNIO
-
-let eventLoop: EventLoop = ...
-let conn = try PostgresConnection.connect(
-    to: .makeAddressResolvingHost("my.psql.server", port: 5432),
-    on: eventLoop
-).wait()
-defer { try! conn.close().wait() }
+  dependencies: [
+    .package(url: "https://github.com/vapor/postgres-nio.git", from: "1.8.0"),
+    ...
+  ]
 ```
 
-Note: These examples will make use of `wait()` for simplicity. This is appropriate if you are using PostgresNIO on the main thread, like for a CLI tool or in tests. However, you should never use `wait()` on an event loop.
-
-There are a few ways to create a `SocketAddress`:
-
-- `init(ipAddress: String, port: Int)`
-- `init(unixDomainSocketPath: String)`
-- `makeAddressResolvingHost(_ host: String, port: Int)`
-
-There are also some additional arguments you can supply to `connect`. 
-
-- `tlsConfiguration` An optional `TLSConfiguration` struct. If supplied, the PostgreSQL connection will be upgraded to use SSL.
-- `serverHostname` An optional `String` to use in conjunction with `tlsConfiguration` to specify the server's hostname. 
-
-`connect` will return a future `PostgresConnection`, or an error if it could not connect. Make sure you close the connection before it deinitializes.
-
-### Authentication
-
-Once you have a connection, you will need to authenticate with the server using the `authenticate` method.
-
+Add `PostgresNIO` to the target you want to use it in:
 ```swift
-try conn.authenticate(
-    username: "your_username",
-    database: "your_database",
-    password: "your_password"
-).wait()
+  targets: [
+    .target(name: "MyFancyTarget", dependencies: [
+      .product(name: "PostgresNIO", package: "postgres-nio"),
+    ])
+  ]
 ```
 
-This requires a username. You may supply a database name and password if needed. 
+#### Creating a connection
 
-### Database Protocol
-
-Interaction with a server revolves around the `PostgresDatabase` protocol. This protocol includes methods like `query(_:)` for executing SQL queries and reading the resulting rows. 
-
-`PostgresConnection` is the default implementation of `PostgresDatabase` provided by this package. Assume `db` here is the connection from the previous example.
+To create a connection, first create a connection configuration object:
 
 ```swift
 import PostgresNIO
 
-let db: PostgresDatabase = ...
-// now we can use client to do queries
+let config = PostgresConnection.Configuration(
+   connection: .init(
+     host: "localhost",
+     port: 5432
+   ),
+   authentication: .init(
+     username: "my_username",
+     database: "my_database",
+     password: "my_password"
+   ),
+   tls: .disable
+)
 ```
 
-### Simple Query
+A connection must be created on a SwiftNIO `EventLoop`. In most server use cases, an 
+`EventLoopGroup` is created at app startup and closed during app shutdown.
```

Comment:
```suggestion
A connection must be created on a SwiftNIO `EventLoop`. In most server use cases, an  `EventLoopGroup` is created at app startup and closed during app shutdown.
```
Extremely minor whitespace formatting nit. Single-line-per-paragraph with word-wrapping usually works better with Markdown for me.